docs/themes.rst

```diff
@@ -240,3 +240,101 @@ And from AMP page:
 
     {# app/themes/<tenant_code>/<theme_name>/amp/amp-theme/index.html.twig #}
     <link rel="canonical" href="{{ url(gimme.article) }}"> {# https://example.com/news/my-articles #}
+
+
+
+Theme Development Workflow
+--------------------------
+
+Theme workflow is based on `Gulp <http://gulpjs.com/>`_ task automation tool.
+There are two main tasks you would use:
+
+- gulp watch
+- gulp build
+
+Watch task is used for development. Run this task when playing with theme css or js.
+Build task should be fired every time before you deploy changes. It minifies your assets and generates service worker.
+
+.. note::
+
+    Please use  `Publisher Theme Starter <https://github.com/tomaszrondio/gulp-service-worker-publisher-setup>`_
+
+Service Worker
+```````````````
+A service worker is a script that your browser runs in the background, separate from a web page, opening the door to features that don't need a web page or user interaction. We use service worker to precache static assets and create runtime cache for images, articles and sections (optional). Goal is to decrease loading time of your page as much as possible.
+
+To learn more about cache and service workers debugging please read:
+`Google: Progressive Web Apps <https://developers.google.com/web/tools/chrome-devtools/progressive-web-apps>`_ ,
+`Google: manage data <https://developers.google.com/web/tools/chrome-devtools/manage-data/local-storage>`_
+
+How to install
+```````````````
+
+If you go with Theme Starter all you have to do is run *gulp build* command. It will generate proper service worker for you.
+
+File structure:
+
+- /public/js/scripts/service-worker-registration.js - registers service worker in browser
```

Comment:
this is about structure in theme starter. Not urls that would work on production